### PR TITLE
#167392596 Pagination Support and Tests

### PR DESCRIPTION
**Pull request by tolumide-ng**
#### Description
Pagination support for fetching Bookmarks, Followers, and Following with Test

#### Type of Change
- [x] New feature (Non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Integration Testing

#### Pivotal Stories
[#167392596](https://www.pivotaltracker.com/story/show/167392596)

### CODE REVIEW

File: src/test/user.spec.js [hunk 1]
```diff
@@ -674,3 +674,140 @@ describe('PATCH /api/v1/auth/user/:id/role', () => {
...
+    const page = 1;
+
+    const res = await server()
+      .get(`${USER_API}/followers?page=${page}&limit=${limit}`)
+      .set('token', token)
+      .send({ followeeId: firstUser.id });
+    expect(res.status).to.equal(200);
+    expect(res.body.message.length).to.equal(numberOfFollowing.count);
+  });
+});
```
Comment (Onyimatics) [inline src/test/user.spec.js:861]: I can see you added pagination on this page. So you need to test the response of the pagination returned on that page. You may need to create more followers so you can assert for the `prev page`, `currentPage`, `next page` and `total number of pages`. You can check the test on [fetch all articles page](https://github.com/andela/ah-kifaru-backend/blob/develop/src/test/article.spec.js). It should be of help.

**Review by Onyimatics — CHANGES_REQUESTED** [review]
Nice work @tolumide-ng. Kindly check the comment I left.

File: src/test/user.spec.js (hunk 1 above)
Comment (tolumide-ng) [reply src/test/user.spec.js:861]: Wow, thank you so much for this @Onyimatics , I would make the changes soon.

**Comment by tolumide-ng** [comment]
> Nice work @tolumide-ng. Kindly check the comment I left.

Changes have been effected, thank you so much😊. Kindly re-review soon.

File: package.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
         "*.js": [
             "npm run lint:fix",
             "npm run prettier",
-            "git add ."
+            "git add .",
+            "npm test"
```
Comment (timi-codes) [inline package.json:36]: Good job. Implementing a precommit hook for running the test files before every commit.

**Review by timi-codes — APPROVED** [review]
Great job @tolumide-ng. I like the way your tests focus on the behavior of the system rather than the implementation details.

**Review by codexempire — APPROVED** [review]
Nice job 👍 implement the asked changes by others

File: src/controllers/article.controller.js [hunk 3]
```diff
@@ -59,21 +59,36 @@ class ArticleController {
...
+        200,
+        null,
+        allBookMarks,
+        paginate.getPageMetadata(count, '/api/v1/articles')
       );
     }
-    return responseGenerator.sendSuccess(res, 200, null, allBookMarks);
+    return responseGenerator.sendError(
+      res,
+      400,
```
Comment (segunolalive) [inline src/controllers/article.controller.js]: The bookmarks list was **found** but it's an `empty` list, therefore it's not a `404`. The `Request` was well-formed, had no erroneous data, therefore it's not a `400 bad request`. 

This is a 200 and should return an empty array

File: src/test/article.spec.js [hunk 4]
```diff
@@ -92,28 +92,42 @@ describe('PATCH api/v1/articles/bookmark', () => {
...
+    );
+    expect(newNumberOfBookmarks.count).to.equal(1);
 
     const token = helper.jwtSigner(firstUser);
+    const page = 1;
+    const limit = 1;
 
     const res = await server()
-      .get(`${ARTICLES_API}/bookmark`)
+      .get(`${ARTICLES_API}/bookmark?page=${page}&limit=${limit}`)
```
Comment (segunolalive) [inline src/test/article.spec.js]: /bookmarks - plural to show that it's a collection.

**Review by segunolalive — COMMENTED** [review]
Good job. Attend to the corrections,

Remove duplicate test assertions.

Then merge

File: src/controllers/article.controller.js (hunk 3 above)
Comment (segunolalive) [reply src/controllers/article.controller.js]:
```
const message = count > 0 ? 'success messge' : 'no bookmarks'
return responseGenerator(res, statusCode, data, message, metadata)

```

Comment (tolumide-ng) [reply src/controllers/article.controller.js]: changes effected

**Pull request merged**